src/blockchains/erc20/lib/add_balances.ts

```diff
@@ -28,66 +28,112 @@ async function doMulticall(web3: Web3, blockNumber: number, addressContract: Uti
-
-    let decodeSuccess = false;
-    if (multicallResult[0]) {
-      try {
-        const decoded: bigint = web3.eth.abi.decodeParameter('uint256', multicallResult[1]) as bigint
-        result.push([blockNumber, address, contractAddress, decoded.toString(36)]);
-        decodeSuccess = true;
+function decodeMulticallResult(addressContractToMulticallResult: Utils.AddressContractToMulticallResult, web3: Web3,
+  blockNumber: number): Utils.BlockNumberAddressContractBalance {
+
+  const addressContract = addressContractToMulticallResult[0]
+  const multicallResult = addressContractToMulticallResult[1]
+
+  if (multicallResult[0]) {
+    try {
+      const decoded: bigint = web3.eth.abi.decodeParameter('uint256', multicallResult[1]) as bigint
+      return [blockNumber, addressContract[0], addressContract[1], decoded.toString(36)];
+    }
+    catch (error: any) {
+      logger.warn(`Error decoding address-contract: ${addressContract[0]}-${addressContract[1]}: ${error}`)
+
+    }
+  }
+
+  logger.warn(`Multicall partial failure for address-contract ${addressContract[0]}-${addressContract[1]}`)
+  return [blockNumber, addressContract[0], addressContract[1], MULTICALL_FAILURE]
+}
+
+async function executeNonBatchMulticall(web3: Web3, addressContracts: Utils.AddressContract[], blockNumber: number)
+  : Promise<Utils.AddressContractToMulticallResult[]> {
+  const allSettled = await Promise.allSettled(addressContracts.flatMap(addressContract =>
+    doMulticall(web3, blockNumber, [addressContract])
+  ))
+
+  if (allSettled.length !== addressContracts.length) {
+    throw new Error(`Multicall response size does not match at block ${blockNumber}`);
+  }
+
+  const errors = allSettled.filter(result => result.status === 'rejected').map(result => result.reason);
+
+  /**
+   * We allow for failed requests at this point. Some responses are just too long and seem like errors. Ex:
+   *
+   * curl -X POST https://ethereum.santiment.net -H "Content-Type: application/json" -d '{
+   * "jsonrpc": "2.0", "id": "1", "method": "eth_call", "params": [
+   *   {
+   *     "to": "0x5BA1e12693Dc8F9c48aAD8770482f4739bEeD696",
+   *     "data": "0xbce38bd70000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000020000000000000000000000000e2975f6a95735bdcad5e39296d37cab6470dacc80000000000000000000000000000000000000000000000000000000000000040000000000000000000000000000000000000000000000000000000000000002470a082310000000000000000000000007ac9368773c8d985c785ddda1964fa7c082cf89b00000000000000000000000000000000000000000000000000000000"
+   *   },
+   *     "0xd70c5d"
+   *   ]
+   *   }'
+   *
+   * This fails on our endpoint but succeeds on Ankr due to different limits. However the response is a long sequence
+   * of 0s.
+   */
+  if (errors.length > 0) {
```

Comment:
Main change in PR, what we did before is a:
1. Try batch multicall.
2. On failure - try get balance per address, on failure - exit.
What we do now is:
1. Try batch multicall.
2. On failure - log warning and continue.